server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -352,6 +364,61 @@ private CompletionStage<RestResponse> report(RestRequest restRequest) {
       );
    }
 
+   private CompletionStage<RestResponse> nodeReport(RestRequest restRequest) {
+      EmbeddedCacheManager cacheManager = invocationHelper.getProtocolServer().getCacheManager();
+      List<Address> members = cacheManager.getMembers();
+      if (members == null) {
+         return report(restRequest);
+      }
+
+      String targetName = restRequest.variables().get("nodeName");
+      ByRef<ByteBuffer> response = new ByRef<>(null);
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      return SecurityActions.getClusterExecutor(cacheManager)
+            .submitConsumer(ecm -> {
+               CacheManagerInfo cmi = ecm.getCacheManagerInfo();
+               if (!cmi.getNodeName().equals(targetName)) return null;
+
+               GlobalComponentRegistry gcr = SecurityActions.getGlobalComponentRegistry(ecm);
+               BasicComponentRegistry bcr = gcr.getComponent(BasicComponentRegistry.class);
+               ServerStateManager ssm = bcr.getComponent(ServerStateManager.class).wired();
+
+               CompletionStage<byte[]> report = ssm.managedServer().getServerReport()
+                     .thenApply(path -> {
+                        try {
+                           return Files.readAllBytes(path);
+                        } catch (IOException e) {
+                           throw new CacheException(e);
+                        }
+                     });
+               return CompletableFutures.uncheckedAwait(report.toCompletableFuture());
+            }, (ignore, info, t) -> {
+               if (t != null) {
+                  throw CompletableFutures.asCompletionException(t);
+               }
+
+               if (info != null) {
+                  response.set(ByteBuffer.wrap(info));
+               }
+            })
+            .thenApply(ignore -> {
+               ByteBuffer report = response.get();
+               if (report == null) {
+                  return responseBuilder.status(NOT_FOUND).build();
+               }
+
+               return responseBuilder
```

Comment:
The responseBuilder code, except the filename, should be the same as that used in `ServerResource#report`. We can create an  method to centralise this logic.